runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/FnApiMonitoringInfoToCounterUpdateTransformer.java

```diff
@@ -35,50 +35,82 @@
 public class FnApiMonitoringInfoToCounterUpdateTransformer
     implements MonitoringInfoToCounterUpdateTransformer {
 
-  final Map<String, MonitoringInfoToCounterUpdateTransformer> counterTransformers = new HashMap<>();
+  final Map<String, Map<String, MonitoringInfoToCounterUpdateTransformer>>
```

Comment:
Can you key this by a tuple instead of a 2 layer dictionary? That might reduce some of the boilerplate below. I believe the Pair class can work here

Reply:
Done.